design-proposals/vm-pool.md

```diff
@@ -0,0 +1,242 @@
+
+## Terms
+
+* **VirtualMachine [VM]** - Refers specifically to the KubeVirt VirtualMachine API object.
+* **VirtualMachinePool [VMPool]** - KubeVirt API for scaling out/in replicas of KubeVirt VirtualMachines.
+* **VirtualMachineConfig [VMConfig]** - KubeVirt API representing a VirtualMachine spec which is referenced by a VirtualMachinePool.
+* **Scale-out and Scale-in** - Terms to describe the action of modifying the replica count of VMs within a VMPool.
+* **Detach VM** - The process of manually separating a VirtualMachine from a VirtualMachinePool.
+* **UpdateIn Strategy** - The policy used to define how a VMPool handles rolling out VMConfig updates to VMs within the pool.
+* **ScaleIn Strategy** - The policy used to define how a VMPool handles removing VMs from a pool during scale-in.
+
+## User Stories
+
+* As a cluster user, I want to automate batch rollout of changes (CPU/Memory/Disk/PubSSHKeys/etc…) across a fleet of VMs.
+* As a cluster user managing fleets of VMs I want to automate scale out of VM instances based on utilization
+* As a cluster user managing fleets of VMs I want to automate scale-in of identical VM instances to optimize cluster resource consumption
+* As a user transitioning workloads to KubeVirt I want to use similar management patterns provided by existing Iaas platforms (AWS, Azure, GCP)
+* As a cluster admin managing nested Kubernetes clusters on top of KubeVirt VMs, I want the ability to elastically scale the underlying KubeVirt VM infrastructure.
+* As a SRE managing the availability of VM fleets I want to automate VM recovery by auto detecting and deleting misbehaving VMs and having the platform spin up fresh new instances as a replacement.
+* As a pool user/manager I want to remove a VM from the pool without modifying it for debugging. The missing VM can be replaced by the pool.
+
+# Design
+
+The VMPool design introduces two new APIs represented as CRDs, the VirtualMachinePool (VMPool) object and the VirtualMachineConfig (VMConfig) object. For the most part, a VMConfig object is simply a VM object without a status section that is used by a VMPool to stamp out multiple replicas.
+
+The figure below illustrates the relationship between a VMPool, VMConfig, and VM. The VMPool stamps out new VM replicas using the contents as in the VMConfig.
+
+
+![VMPool+VMConfig+VM](vm-pool-images/figure1.png "VMPool and VMConfig relationship")
+
+In order to make this even more tangible, below is a figure that outlines the types of tunings present in each of these APIs.
+
+![VMPool vs. VMConfig](vm-pool-images/figure2.png "VMPool vs. VMConfig")
+
+## VirtualMachineConfig (VMConfig) API
+
+The VMConfig API contains a VirtualMachineSpec structure that is identical to a VM object, which makes it similar to a VM object itself. However there are some key differences here between the two objects.
```

Comment:
Asked in SIG-scale:  Why do we want to use a VM object in the Pool and not VMIs?  In other words, a Pool is an abstraction for many objects, why can't those be VMIs instead of VMs? 

My take away was that VMs (the kubevirt API) hold state.  A VMPool is an abstraction for stateful virtual machines or those that need advanced management.  VMI replicaset would be the abstraction for stateless virtual machines.  Do you want to add a comment about this in case it comes up again?